data/txHandler.go

```diff
@@ -221,16 +223,27 @@ func (handler *TxHandler) asyncVerifySignature(arg interface{}) interface{} {
 }
 
 func (handler *TxHandler) processIncomingTxn(rawmsg network.IncomingMessage) network.OutgoingMessage {
+	// attempt to preallocate buffer on stack:
+	// hardcode maxTxGroupSize constant and use backing arrays,
+	// or fallback to heap allocation it MaxTxGroupSize is greater than our constant
+	const maxTxGroupSize = 16 // no ledger lookup and allocate the backing array on stack
```

Comment:
if this moves out to top level we could make a unit test to check against current protocol and vFuture

Reply:
done